openedx/features/course_duration_limits/access.py

```diff
@@ -130,12 +131,30 @@ def register_course_expired_message(request, course):
             )
         )
     else:
-        upgrade_message = _('Your access to this course expires on {expiration_date}. \
-                    {a_open}Upgrade now {sronly_span_open}to retain access past {expiration_date}.\
-                    {span_close}{a_close}{sighted_only_span_open}for unlimited access.{span_close}')
+        enrollment = CourseEnrollment.get_enrollment(request.user, course.id)
+        if enrollment is None:
+            return
+
+        upgrade_deadline = enrollment.upgrade_deadline
+        if upgrade_deadline is None:
+            return
+
+        expiration_message = _('{div_open}{strong_open}Audit Access Expires {expiration_date}{strong_close}{div_close}'
+                               '{div_open}You lose all access to this course, including your progress, on '
+                               '{expiration_date}.{div_close}')
+        upgrade_deadline_message = _('Upgrade by {upgrade_deadline} to get unlimited access to the course as long as '
+                                     'it exists on the site.')
+        upgrade_link = _('{a_open}Upgrade now{sronly_span_open} to retain access past '
+                         '{expiration_date}{span_close}{a_close}')
+
+        full_message = expiration_message
+        if timezone.now() < upgrade_deadline:
```

Comment:
so yeah it looks like we didn't already have this state. I asked Darren if he had requirements already for this state